v0/destinations/mailchimp/transform.js

```diff
@@ -115,141 +161,169 @@ async function responseBuilderSimple(payload, message, mailChimpConfig) {
-      case destinationConfigKeys.dataCenterId:
-        mailChimpConfig.dataCenterId = `${destination.Config[key]}`;
-        break;
-      case "enableMergeFields":
-        mailChimpConfig.enableMergeFields =
-          destination.Config.enableMergeFields;
-        break;
-      default:
-        logger.debug("MailChimp: Unknown key type: ", key);
-        break;
-    }
-  });
+  return responseBuilderSimple(
+    mergedFieldPayload,
+    message,
+    Config,
+    audienceId,
+    emailExists
+  );
+};
 
-  const mailChimpExists = get(message, "context.MailChimp");
-  if (mailChimpExists) {
-    const listIdExists = get(message, "context.MailChimp.listId");
-    if (listIdExists) {
-      mailChimpConfig.audienceId = message.context.MailChimp.listId;
-    }
+const process = async event => {
+  const { message, destination } = event;
+
+  const destConfig = destination.Config;
+  if (!destConfig.apiKey) {
+    throw new CustomError("API Key not found. Aborting", 400);
   }
-  return mailChimpConfig;
-}
 
-async function processIdentify(message, destination) {
-  const mailChimpConfig = getMailChimpConfig(message, destination);
-  const properties = await getTransformedJSON(message, mailChimpConfig);
-  return responseBuilderSimple(properties, message, mailChimpConfig);
-}
+  if (!destConfig.audienceId) {
+    throw new CustomError("Audience Id not found. Aborting", 400);
+  }
+
+  if (!destConfig.datacenterId) {
+    throw new CustomError("DataCenter Id not found. Aborting", 400);
+  }
+
+  if (!message.type) {
+    throw new CustomError(
+      "Message Type is not present. Aborting message.",
+      400
+    );
+  }
+
+  const messageType = message.type.toLowerCase();
 
-async function processSingleMessage(message, destination) {
-  if (message.type !== EventType.IDENTIFY) {
-    throw new CustomError(`message type ${message.type} is not supported`, 400);
+  let response;
+  switch (messageType) {
+    case EventType.IDENTIFY:
+      response = await identifyResponseBuilder(message, destination);
+      break;
+    default:
+      throw new CustomError(
+        `message type ${messageType} is not supported`,
+        400
+      );
   }
+  return response;
+};
+
+function batchEvents(arrayChunks) {
+  // Batching reference doc: https://mailchimp.com/developer/marketing/api/lists/
+  const batchedResponseList = [];
+
+  // list of chunks [ [..], [..] ]
+  arrayChunks.forEach(chunk => {
+    const batchResponseList = [];
+    const metadata = [];
+
+    // extracting destination
+    // from the first event in a batch
+    const { destination, message } = chunk[0];
+
+    let batchEventResponse = defaultBatchRequestConfig();
 
-  return processIdentify(message, destination);
+    // Batch event into dest batch structure
+    chunk.forEach(ev => {
+      batchResponseList.push(ev.message.body.JSON);
+      metadata.push(ev.metadata);
+    });
+
+    batchEventResponse.batchedRequest.body.JSON = {
+      members: batchResponseList
```

Comment:
Do we have only members parameters for batch? Let's put a quick reference to the Mailchimp doc for batching.

Reply:
[Here](https://mailchimp.com/developer/marketing/api/lists/batch-subscribe-or-unsubscribe/), in Mailchimp docs, we do have two other parameters for batch. They are 
`sync_tags`: Whether this batch operation will replace all existing tags with tags in request.
`update_existing`: Whether this batch operation will change existing members' subscription status.

For `update_existing`, I tried sending data to the endpoint directly from the postman. This is the observation:

1. When an existing user is called it throws an error(displayed below), if `update_existing` is `false` or not sent.

<img width="1074" alt="image" src="https://user-images.githubusercontent.com/58467752/172159089-9a5b06f0-9262-402a-b6c6-b58eb86c5897.png">

2. When `update_existing` is `true`, it updates the user details.

<img width="1041" alt="image" src="https://user-images.githubusercontent.com/58467752/172159660-46bfd46c-4235-4a69-9b2d-0133da86e97c.png">

Should we set `update_existing` as `true` by default? (Assuming to update user details, when a user exists)

Reply:
We should support that also we should set forth this in docs. We need to update docs in terms of batching cc @shrouti1507